assets/javascripts/discourse/initializers/chat-setup.js

```diff
@@ -129,6 +132,82 @@ export default {
           }
         });
       });
+      api.addSidebarSection((BaseSectionHeader, BaseSectionLink) => {
+        return class extends BaseSectionHeader {
+          @tracked sectionLinks = A([]);
+
+          constructor() {
+            super(...arguments);
+            this.chatService = container.lookup("service:chat");
+            this.chatService.getChannels().then((channels) => {
+              channels.publicChannels.forEach((channel) => {
+                this.sectionLinks.pushObject(
+                  new (class extends BaseSectionLink {
+                    get name() {
+                      return channel.chatable_id;
+                    }
+                    get route() {
+                      return "chat.channel";
+                    }
+                    get model() {
+                      return {
+                        ...channel,
+                        channelId: channel.id,
+                        channelTitle: channel.title,
+                      };
+                    }
+                    get title() {
+                      return channel.title;
+                    }
+                    get text() {
+                      return channel.title;
+                    }
+                  })()
```

Comment:
We don't really need to define the class over and over again here. We can create a generic class and pass the `channel` in when initialising the class.